BEIMA.Client/src/pages/Devices/DevicePage.js

```diff
@@ -0,0 +1,382 @@
+      const value = event.target.value
+      if(target === 'deviceNotes'){
+        setNotes(value)
+      } else if (target === 'deviceTag'){
+        setTag(value)
+      } else if (target === 'deviceManufacturer'){
+        setManufacturer(value)
+      } else if (target === 'deviceModelNumber'){
+        setModelNum(value)
+      } else if (target === 'deviceSerialNumber'){
+        setSerialNum(value)
+      } else if (target === 'deviceYearManufactured'){
+        setYearManufactured(value)
+      } else if (target === 'locationNotes'){
+        setLocNotes(value)
+      } else if (target === 'deviceLatitude'){
+        setLat(value)
+      } else if (target === 'deviceLongitude'){
+        setLong(value)
+      } else if (target === 'deviceBuildingId'){
+        setBuildingId(value)
+      }
+    }
+
+    const onCustomFieldChange = (event) => {
+      const target = event.target.id
+      const value = event.target.value
+      let temp = {...fields}
+      temp[target] = value
+      setFields(temp)
+    }
+
+    const onImageChange = (event) => {
+      if(event.target.files && event.target.files[0]){
+        setImageCopy(URL.createObjectURL(event.target.files[0]))
+      }
+    }
+
+    const onDocumentChange = (event) => {
+      var files = event.target.files
+      if(files && files.length > 0){
+        var tempAdded = [...addedDocs]
+        var temp = [...files]
+        temp.forEach(file => {
+          tempAdded.push(file)
+        })
+        setAddedDocs(tempAdded)
+      }
+    }
+
+    const deleteDocument = (doc) => {
+      var tempDocs = [...docCopy]
+      var tempAdded = [...addedDocs]
+      var tempDel = [...removedDocs]
+
+      if(tempDocs.includes(doc)){
+        tempDocs = tempDocs.filter(val => val !== doc)
+        tempDel.push(doc)
+        setDocCopy(tempDocs)
+        setRemovedDocs(tempDel)
+      } else {
+        tempAdded = tempAdded.filter(val => val.name !== doc)
+        setAddedDocs(tempAdded)
+      }
+    }
+
+    return (
+      <Form className={styles.form}>
+        <Form.Group className="mb-3">
+          {editable ? 
+           <div className={styles.buttonRow}>
+              <Button onClick={submit}>
+                Save
+              </Button>
+              <Button variant="secondary" onClick={cancel}>
+                Cancel
+              </Button>
+           </div>
+          : 
+            <Button variant="primary" id="editbtn" onClick={() => setEditable(true)}>
+              Edit
+            </Button>
+          }
+        </Form.Group>
+
+        <Form.Group>
+          <Form.Label><b>Device Image</b></Form.Label>
+        </Form.Group>
+
+        <Form.Group className={[styles.image, "mb-3"].join(' ')} id="imageDisplay">
+          <Card>
+            <Card.Body>
+              <Image src={imageCopy} fluid/>
+            </Card.Body>
+          </Card>
+        </Form.Group>
+
+        <ImageUpload type="Device Image" editable={editable} onImageChange={onImageChange}/>
```

Comment:
I think this should be hidden when we are in the viewing state.

Reply:
so maybe use something like hidden={!editable} or something

Reply:
I'm good either way, but I'm leaning towards keeping them hidden when in the viewing state.

Reply:
Updated it to be hidden when not in an editing state